python/ray/train/lightning/lightning_utils.py

```diff
@@ -0,0 +1,117 @@
+import os
+import logging
+import torch
+from torch import Tensor
+from typing import Any, Dict, Optional
+from copy import deepcopy
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning.utilities import rank_zero_info, rank_zero_only
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+from pytorch_lightning.strategies import DDPStrategy
+from lightning_fabric.plugins.environments.lightning import LightningEnvironment
+
+import ray
+from ray.air import session
+from ray.train.lightning.lightning_checkpoint import LightningCheckpoint
+
+
+logger = logging.getLogger(__name__)
+
+class RayModelCheckpoint(ModelCheckpoint):
+    """AIR customized ModelCheckpoint callback """
```

Comment:
Can we add some more description about how this works? What assumptions are made?

Ex: if reporting happens every X batches, checkpointing must happen every n*X.

Reply:
Yeah I'll add some descriptions. 
We don't have this checkpoint frequency assumption now. We always use the latest reported metrics for the checkpoint.